Exploration_SampleEntropy_IMU.py

```diff
@@ -0,0 +1,448 @@
+            HS.append(jj-window+HS_idx)
+            jj = jj+500
+        jj = jj+1
+            
+    # Compute the mid-stance indicies
+    MS = np.array([(np.argmin(MS_sig[HS[jj]+10:HS[jj]+int((HS[jj+1]-HS[jj])*0.2)])+HS[jj]+10)  for jj in range(len(HS)-1)]) 
+    # MS = []
+    # for jj in range(len(HS)-1):
+    #     print(jj)
+    #     MS.append(np.argmin(MS_sig[HS[jj]+10:HS[jj]+int((HS[jj+1]-HS[jj])*0.2)])+HS[jj]+10)
+    HS = np.array(HS[:-1])
+        
+    
+    return [HS,MS]
+
+def findRotToLab(accel_vec):
+    """
+    Function to find the rotation of the foot flat acceleration vector to the 
+    defined lab gravity coordinate system.
+
+    Parameters
+    ----------
+    accel_vec : numpy array
+        3x1 vector of the x,y,z acceleration at foot flat
+
+    Returns
+    -------
+    theta : numpy array
+        3x1 vector for the rotation from the foot flat acceleration to the lab
+        coordinate system
+
+    """
+    iGvec = accel_vec
+    iGvec = iGvec/np.linalg.norm(iGvec)
+    # Define the lab coordinate system gravity vector
+    lab_Gvec = np.array([1,0,0]).T
+    #______________________________________________________________________
+    # Compute the rotation matrix
+    C = np.cross(lab_Gvec,iGvec)
+    D = np.dot(lab_Gvec,iGvec)
+    Z = np.array([[0,-C[2],C[1]],[C[2],0,-C[0]],[-C[1],C[0],0]])
+    # Note: test the rotation matrix (R) that the norm of the colums and the rows is 1
+    R = np.eye(3)+Z+(Z@Z)*(1-D)/np.linalg.norm(C)**2
+    # Compute the rotation angles
+    theta = np.zeros(3) # Preallocate
+    theta[1] = arctan2(-R[2,0],np.sqrt(R[0,0]**2+R[1,0]**2))
+    # Conditional statement in case theta[1] is +/- 90 deg (pi/2 rad)
+    if theta[1] == np.pi/2:
+        theta[2] = 0
+        theta[0] = arctan2(R[0,1],R[1,1])
+    elif theta[1] == -np.pi/2:
+        theta[2] = 0
+        theta[0] = -arctan2(R[0,1],R[1,1])
+    else: 
+        theta[2] = arctan2(R[1,0]/cos(theta[1]),R[0,0]/cos(theta[1]))
+        theta[0] = arctan2(R[2,1]/cos(theta[1]),R[2,2]/cos(theta[1]))
+    theta = theta*180/np.pi
+    return theta
+
+def rot2gravity(timeseries, HS, MS, GS):
+    """
+    Function to find the rotation of the foot flat acceleration vector to the 
+    defined lab gravity coordinate system.
+
+    Parameters
+    ----------
+    accel_vec : numpy array
+        3x1 vector of the x,y,z acceleration at foot flat
+
+    Returns
+    -------
+    theta : numpy array
+        3x1 vector for the rotation from the foot flat acceleration to the lab
+        coordinate system
+
+    """
+    #timeseries = up
+    
+    MS_frames = 20
+    
+    GS = GS[0:-2]
+    
+    
+    for count,jj in enumerate(GS[:-1]):
+        # Obtain the acceleration and gyroscope from foot contact (HS) to the
+        # subsiquent midstance (MS). MS is needed for linear drift correction
+        
+        #count = 0
+        #jj = GS[count]
+        
+        jjn = GS[count +1]
+        acc_stride_plus = timeseries[HS[jj]:MS[jjn],1:4]
```

Comment:
I think that you have the accelerometer and gyroscope signals flipped here in lines 245 and 246. On lines 376, 383, and 390 the gyro is concatenated prior to the accelerometer